### PR TITLE
Fixed PST Flow Regulation on non impedant branch causing exception

### PR DESCRIPTION
**Pull request by jeandemanged**
**Please check if the PR fulfills these requirements**

- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem?**
No

**What kind of change does this PR introduce?**
Bugfix

**What is the current behavior?**
Exception thrown if a PST has flow regulation on a terminal of a zero impedance branch, in mode `LowImpedanceBranchMode.REPLACE_BY_ZERO_IMPEDANCE_LINE` (default).

**Does this PR introduce a breaking change or deprecate an API?**
No

**Other information**:

Possible resolution: we might want to keep such branches as low impedance.

```
Caused by: java.util.NoSuchElementException: No value present
	at java.base/java.util.Optional.orElseThrow(Optional.java:382)
	at com.powsybl.openloadflow.ac.equations.AcEquationSystemCreator.updateTransformerPhaseControlEquations(AcEquationSystemCreator.java:474)
	at com.powsybl.openloadflow.ac.equations.AcEquationSystemUpdater.onTransformerPhaseControlChange(AcEquationSystemUpdater.java:43)
	at com.powsybl.openloadflow.network.LfNetworkListenerTracer.onTransformerPhaseControlChange(LfNetworkListenerTracer.java:54)
	at com.powsybl.openloadflow.network.impl.AbstractLfBranch.setPhaseControlEnabled(AbstractLfBranch.java:163)
	at com.powsybl.openloadflow.ac.outerloop.PhaseControlOuterLoop.initialize(PhaseControlOuterLoop.java:39)
	at com.powsybl.openloadflow.ac.outerloop.PhaseControlOuterLoop.initialize(PhaseControlOuterLoop.java:24)
	at com.powsybl.openloadflow.ac.AcloadFlowEngine.run(AcloadFlowEngine.java:130)
```

### CODE REVIEW

File: src/main/java/com/powsybl/openloadflow/network/impl/AbstractLfBranch.java [hunk 1]
```diff
@@ -281,6 +281,10 @@ public void setMinZ(double lowImpedanceThreshold) {
             if (piModel.setMinZ(lowImpedanceThreshold, loadFlowModel)) {
                 LOGGER.trace("Branch {} has a low impedance in {}, set to min {}", getId(), loadFlowModel, lowImpedanceThreshold);
                 zeroImpedanceContextByModel.get(loadFlowModel).zeroImpedance = false;
+            } else if (isZeroImpedance(loadFlowModel)) {
```
Comment (annetill) [inline src/main/java/com/powsybl/openloadflow/network/impl/AbstractLfBranch.java:284]: I am going to add a unit test for this fix.

File: src/test/java/com/powsybl/openloadflow/ac/AcLoadFlowPhaseShifterTest.java [hunk 2]
```diff
@@ -157,6 +157,24 @@ void remoteFlowControlT2wtTest() {
         assertEquals(0, t2wt.getPhaseTapChanger().getTapPosition());
     }
 
+    @Test
+    void remoteFlowControlOnZeroImpedanceBranch() {
```
Comment (annetill) [inline src/test/java/com/powsybl/openloadflow/ac/AcLoadFlowPhaseShifterTest.java:161]: @jeandemanged indeed I can remove your test now as I have fixed an existing one...

File: src/main/java/com/powsybl/openloadflow/network/LfNetwork.java [hunk 3]
```diff
@@ -482,8 +482,9 @@ public void fix(boolean minImpedance, double lowImpedanceThreshold) {
             }
         } else {
             // zero impedance controller phase shifter is not supported
+            // zero impedance controlled branch is not supported
```
Comment (jeandemanged) [inline src/main/java/com/powsybl/openloadflow/network/LfNetwork.java:485]:
```suggestion
            // zero impedance controller or controlled branch is not supported
```

**Review by jeandemanged — COMMENTED** [review]
thanks a lot @annetill, that's perfect

**Pull request merged**